loadbalancer/algorithm.go

```diff
@@ -309,13 +213,14 @@ func (ch *consistentHash) Apply(ctx *routing.LBContext) routing.LBEndpoint {
 		return ctx.Route.LBEndpoints[0]
 	}
 
-	choice := ch.chooseConsistentHashEndpoint(ctx, noSkippedEndpoints)
-
-	if ctx.Route.LBFadeInDuration <= 0 {
-		return ctx.Route.LBEndpoints[choice]
-	}
+	ctx.Route.Lock()
+	defer ctx.Route.Unlock()
 
-	return withFadeIn(ch.rand, ctx, ch.notFadingIndexes, choice, ch)
+	choice := ch.chooseConsistentHashEndpoint(ctx, func(i int) bool {
+		_, ok := ctx.Route.GoodHosts[ctx.Route.LBEndpoints[i].Host]
```

Comment:
Here is it not clear why we use Host instead of treating endpoint string as opaque value.